src/librustdoc/html/render/search_index.rs

```diff
@@ -207,7 +207,7 @@ impl SerializedSearchIndex {
         path_data: Option<PathData>,
         entry_data: Option<EntryData>,
         desc: String,
-        function_data: Option<FunctionData>,
+        function_data: Option<IndexItemFunctionType>,
```

Comment:
this is the (hopefully) impactful part: `FunctionData` is a OTW representation of `IndexItemFunctionType`, and there's no need to do the writing and parsing part until we leave the boundaries of the current process invocation